src/core/thread/mle_router.cpp

```diff
@@ -1739,11 +1735,25 @@ void MleRouter::HandleStateUpdateTimer(void)
         ExitNow();
 
     case kDeviceStateChild:
-        if (routerStateUpdate && GetActiveRouterCount() < mRouterUpgradeThreshold)
+        if (routerStateUpdate)
         {
-            // upgrade to Router
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
-            ExitNow();
+            if (GetActiveRouterCount() < mRouterUpgradeThreshold)
+            {
+                // upgrade to Router
+                BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+                ExitNow();
+            }
+            else
```

Comment:
Should the random jitter for REED occur only once after becoming a Child?  More concretely, should this be `else if (!mAdvertiseTimer.IsRunning())`?  Without the additional check, any time the REED evaluates whether or not to switch to Router (i.e. as part of receiving an MLE Advertisement) will cause the REED to also schedule a MLE Advertisement transmission with a random jitter.